project/Build.scala

```diff
@@ -1065,83 +1065,105 @@ object Build extends sbt.Build {
 
   // Testing
 
-  val testTagSettings = Seq(
-      testOptions in Test ++= {
-        @tailrec
-        def envTagsFor(env: JSEnv): Seq[Tests.Argument] = env match {
-          case env: RhinoJSEnv =>
-            val baseArgs = Seq("-trhino")
-            val args =
-              if (env.sourceMap) baseArgs :+ "-tsource-maps"
-              else baseArgs
-
-            Seq(Tests.Argument(args: _*))
-
-          case env: NodeJSEnv =>
-            val baseArgs = Seq("-tnodejs", "-ttypedarray")
-            val args = {
-              if (env.sourceMap) {
-                if (!env.hasSourceMapSupport) {
-                  val projectId = thisProject.value.id
-                  sys.error("You must install Node.js source map support to " +
-                    "run the full Scala.js test suite (npm install " +
-                    "source-map-support). To deactivate source map " +
-                    s"tests, do: set jsEnv in $projectId := " +
-                    "NodeJSEnv().value.withSourceMap(false)")
-                }
-                baseArgs :+ "-tsource-maps"
-              } else
-                baseArgs
-            }
+  val testTagSettings = {
+    class OptionTags(stage: Stage, sems: Semantics, jsEnv: JSEnv, projectId: String) {
```

Comment:
Looks like this could simply be a `def makeOptionTags(stage, ...): Seq[String]`. (and have callers call `map` instead of `as`)